src/app/core/services/label.service.ts

```diff
@@ -54,59 +77,111 @@ export class LabelService {
   }
 
   /**
-   * Returns the color of the label using the label-color mapping
+   * Returns the color of the label by iterating through the list of
+   * available labels.
    * @param labelValue: the label's value (e.g Low / Medium / High)
-   * @return a string with the color code of the label, or white color if
-   * no labelValue was provided or no such mapping was found
+   * @return a string with the color code of the label or a white color if
+   * no labelValue was provided or the color
    */
   getColorOfLabel(labelValue: string): string {
-    const color = this.labelColorMap.get(labelValue);
+    const WHITE_COLOR = 'ffffff';
+    if (labelValue === '') {
+      return WHITE_COLOR;
+    }
+
```

Comment:
Perhaps you can concat the label arrays to create a single array for easy search? https://www.tutorialspoint.com/typescript/typescript_array_concat.htm

Reply:
Would it be okay to use the `for` loop to concatenate the Label arrays? If we add a new list in the future, we wouldn't have to change the code here to account for it.

Reply:
Good point. You can try to abstract it out too e.g., write a util method getRequiredLabelsAsArray() that loops over the map and returns an array